### PR TITLE
[CI] Replace manylinux_2_24 image

### PR DESCRIPTION
**Pull request by phschiele**
## Description
Originally, we chose `manylinux_2_24` due to compatibility issues with `manylinux2014` (see comment [here](https://github.com/cvxpy/cvxpy/pull/1378#issue-895729469)).
With the image potentially reaching EOL soon (https://github.com/pypa/manylinux/issues/1332), this PR removes the hardcoding of the `manylinux_2_24` image, which currently results in `manylinux2014` being chosen. I've upgraded the version of `cibuildwheel`, but I'm not sure if this solved the original problem, or just that potentially problematic dependencies have been upgraded in the meantime.
See the outputs of [this run](https://github.com/phschiele/cvxpy/actions/runs/2456407768) to verify.

I've tested that the produced wheels can be installed on my local ubuntu 20.04 machine. 

Note: From the repos where we have deployed a similar setup, only [ecos-python](https://github.com/embotech/ecos-python/blob/master/.github/workflows/build.yml) still has `manylinux_2_24` fixed. I.e., [scs-python](https://github.com/bodono/scs-python/blob/master/.github/workflows/build.yml) and [diffcp](https://github.com/cvxgrp/diffcp/blob/master/.github/workflows/build.yml) are fine.

One side effect might be that rerunning the CI to upload the M1 wheels as discussed with @rileyjmurray would also upload the `manylinux2014` wheels alongside the existing `manylinux_2_24` wheels if this PR here is merged first. That should hopefully not cause issues, but we can just merge and deploy the M1 wheels before this one if we want to prevent it.

## Type of change
- [ ] New feature (backwards compatible)
- [ ] New feature (breaking API changes)
- [ ] Bug fix
- [x] Other (Documentation, CI, ...)

## [Contribution checklist](https://www.cvxpy.org/contributing/index.html#contribution-checklist)
- [ ] Add our license to new files.
- [ ] Check that your code adheres to our coding style.
- [ ] Write unittests.
- [ ] Run the unittests and check that they’re passing.
- [ ] Run the benchmarks to make sure your change doesn’t introduce a regression.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -144,7 +142,7 @@ jobs:
         run: |
           python -m pip install --upgrade twine
           twine check wheelhouse/*
-          twine upload --repository-url $PYPI_SERVER wheelhouse/* -u $PYPI_USER -p $PYPI_PASSWORD
+          twine upload --repository-url $PYPI_SERVER wheelhouse/* -u $PYPI_USER -p $PYPI_PASSWORD
```
Comment (phschiele) [inline .github/workflows/build.yml:145]:
```suggestion
          twine upload --repository-url $PYPI_SERVER wheelhouse/* -u $PYPI_USER -p $PYPI_PASSWORD
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -63,7 +63,6 @@ jobs:
       - name: Set Additional Envs
         run: |
           echo "PYTHON_SUBVERSION=$(echo $PYTHON_VERSION | cut -c 3-)" >> $GITHUB_ENV
-          echo "DEPLOY=$( [[ $GITHUB_EVENT_NAME == 'push' && $GITHUB_REF == 'refs/tags'* ]] && echo 'True' || echo 'False' )" >> $GITHUB_ENV
```
Comment (phschiele) [inline .github/workflows/build.yml:66]: This is unrelated, the `DEPLOY` environment variable was unused in this job.

**Review by SteveDiamond — APPROVED** [review]

**Comment by SteveDiamond** [comment]
Thanks @phschiele! Let's upload the M1 wheels first.
